src/react-components/input/ToolbarMicButton.js

```diff
@@ -0,0 +1,50 @@
+import React, { useRef, useCallback } from "react";
+import PropTypes from "prop-types";
+import { FormattedMessage } from "react-intl";
+import { ToolbarButton } from "../input/ToolbarButton";
+import { useVolumeMeter } from "../misc/useVolumeMeter";
+
+export function ToolbarMicButton({ scene, ...rest }) {
+  const ref = useRef();
+  const setRef = useCallback(
+    node => {
+      ref.current = node;
+    },
+    [ref]
+  );
+  const update = useCallback(
+    level => {
+      const node = ref.current;
+      if (node) {
+        const rect = node.querySelector("rect");
+        if (level <= 0.1) {
+          rect.setAttribute("height", 0);
+        } else if (level < 0.3) {
+          rect.setAttribute("y", 8);
+          rect.setAttribute("height", 4);
+        } else {
+          rect.setAttribute("y", 4);
+          rect.setAttribute("height", 8);
+        }
+      }
+    },
+    [ref]
+  );
+  useVolumeMeter({
+    analyser: scene.systems["hubs-systems"].audioSystem.outboundAnalyser,
+    update
+  });
+  return (
+    <ToolbarButton
+      ref={setRef}
```

Comment:
Pretty sure `setRef` is redundant. You can use `ref` directly here.
```suggestion
      ref={ref}
```

Reply:
Oh yep that would be only necessary in case of using an effect but this is just a timed callback so it should be fine. 

Thinking about your later comment about removing the rect query, I think we can probably keep this and use it for gathering the rect once the node is ready to avoid the repeated querySelector.